src/web/components/Onwards/Carousel/Carousel.tsx

```diff
@@ -483,14 +499,21 @@ export const Carousel: React.FC<OnwardsType> = ({
 	};
 
 	useEffect(() => {
-		if (carouselRef.current) {
-			carouselRef.current.addEventListener(
+		const carousel = carouselRef.current;
+		if (carousel) {
+			carousel.addEventListener('scroll', libDebounce(getSetIndex, 100));
+			return carousel.removeEventListener(
 				'scroll',
 				libDebounce(getSetIndex, 100),
 			);
 		}
 	});
 
+	// No idea if this is the best approach but it prevents issues with libDebounce
+	// using old data to determine the max index. Instead we say update maxIndex
+	// when index changes and compare it against the prior maxIndex only.
+	useEffect(() => setMaxIndex((m) => Math.max(index, m)), [index]);
```

Comment:
I wasn't sure I understood this change. Is it because we dynamically load more elements to the carousel/infinite scroll or something?

Reply:
It's to keep track of how far along the user scrolled. They might get all the way to the end in which case `maxIndex` will be 10 or whatever, but it's more likely we'll see people scrolling once or twice. It helps to know if people are going backwards and forwards before selecting an article.

Reply:
Ah okay cool.